src/masternode.cpp

```diff
@@ -491,31 +414,26 @@ bool CMasternodeBroadcast::Create(CTxIn txin, CService service, CKey keyCollater
              CBitcoinAddress(pubKeyCollateralAddressNew.GetID()).ToString(),
              pubKeyMasternodeNew.GetID().ToString());
 
-
-    CMasternodePing mnp(txin);
-    if(!mnp.Sign(keyMasternodeNew, pubKeyMasternodeNew)) {
-        strErrorRet = strprintf("Failed to sign ping, masternode=%s", txin.prevout.ToStringShort());
+    auto Log = [&strErrorRet,&mnbRet](std::string sErr)->bool
+    {
+        strErrorRet = sErr;
         LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
         mnbRet = CMasternodeBroadcast();
         return false;
-    }
+    };
+
+    CMasternodePing mnp(txin);
+    if (!mnp.Sign(keyMasternodeNew, pubKeyMasternodeNew))
+        return Log(strprintf("Failed to sign ping, masternode=%s", txin.prevout.ToStringShort()));
 
     mnbRet = CMasternodeBroadcast(service, txin, pubKeyCollateralAddressNew, pubKeyMasternodeNew, PROTOCOL_VERSION);
 
-    if(!mnbRet.IsValidNetAddr()) {
-        strErrorRet = strprintf("Invalid IP address, masternode=%s", txin.prevout.ToStringShort());
-        LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
-        mnbRet = CMasternodeBroadcast();
-        return false;
-    }
+    if (!mnbRet.IsValidNetAddr())
+        return Log(strprintf("Invalid IP address, masternode=%s", txin.prevout.ToStringShort()));
 
     mnbRet.lastPing = mnp;
-    if(!mnbRet.Sign(keyCollateralAddressNew)) {
-        strErrorRet = strprintf("Failed to sign broadcast, masternode=%s", txin.prevout.ToStringShort());
-        LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
-        mnbRet = CMasternodeBroadcast();
-        return false;
-    }
+    if (!mnbRet.Sign(keyCollateralAddressNew))
+        return Log(strprintf("Failed to sign broadcast, masternode=%s", txin.prevout.ToStringShort()));
```

Comment:
I think that the behaviour is the same as before.
The `mnbRet = CMasternodeBroadcast()` is included in the `Log` local (lambda) function
(I guess that's what you mean by invalidation).

This change adds a small benefit - it makes it slightly easier to check the cases when we do come to refactor properly. The local function cuts out a bit of repetition, and so compresses the code to scan for the cases, at the expense of having to check what the function is doing.

Really, though, I want to get rid of these `Create` functions and replace with regular constructors, adding a flag in the class for 'invalid' state if needed. Do you think that is feasible, or is there a need for a separate Create function I'm missing?